File: October_5/Solution.js
var addOneRow = function(root, val, depth) {
  if (depth === 1) {
    return new TreeNode(val, root, null)
  }
  
  dfs(root, val, depth, 1)
  return root
};

var dfs = function(node, val, depth, currentDepth) {
    if (!node) return;
  
    if (currentDepth + 1 === depth) {
      node.left = new TreeNode(val, node.left, null)
      node.right = new TreeNode(val, null, node.right)
    }
  
    dfs(node.left, val, depth, currentDepth + 1)
    dfs(node.right, val, depth, currentDepth + 1)
}

File: October_5/solution.java
public TreeNode addOneRow(TreeNode root, int val, int depth) {
            if(depth == 1) return new TreeNode(val,root,null);
            addOneRow_helper(root,val,1,depth);
			return root;
        }
private void addOneRow_helper(TreeNode node, int val, int currDepth, int depth) {
            if(node == null || currDepth >= depth) return;
            if(currDepth + 1 == depth) {
                node.left = new TreeNode(val,node.left,null);
                node.right = new TreeNode(val,null,node.right);
                return;
            }
            else {
                addOneRow_helper(node.left,val,currDepth+1,depth);
                addOneRow_helper(node.right,val,currDepth+1,depth);
            }
        }
